Commit may get cancelled by an error from Prettier if there isn't internet connection

```
npm ERR! code ENOTFOUND
npm ERR! syscall getaddrinfo
npm ERR! errno ENOTFOUND
npm ERR! network request to https://registry.npmjs.org/prettier failed, reason: getaddrinfo ENOTFOUND registry.npmjs.org
npm ERR! network This is a problem related to network connectivity.
npm ERR! network In most cases you are behind a proxy or have bad network settings.
npm ERR! network 
npm ERR! network If you are behind a proxy, please make sure that the
npm ERR! network 'proxy' config is set properly.  See: 'npm help config'
npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Dev\AppData\Local\npm-cache\_logs\2022-11-18T18_30_52_044Z-debug-0.log
```

--- a/pre-commit
+++ b/pre-commit
@@ -55,8 +55,14 @@ runPrettier () {
         return 0
     fi
 
-    # Run prettier over all Javascript files
-    npx --yes prettier --ignore-unknown --write AnkiDroid/**/*.js
+    CHANGED_JS_FILES="$(git --no-pager diff --name-status --no-color --cached | awk '$1 != "D" && $2 ~ /\.js/ { print $2 }')"
+        if [ -z "$CHANGED_JS_FILES" ]; then
+            echo "No Javascript staged files. Hence, skipping pre-commit Prettier run."
+            return 0
+        fi;
+
+    # Prettify changed files
+    echo "$CHANGED_JS_FILES" | xargs npx prettier --ignore-unknown --write
 
     echo "Completed npx prettier run."
     echo "$CHANGED_JS_FILES" | while read -r file; do

--- a/pre-commit
+++ b/pre-commit
@@ -67,5 +67,5 @@ runPrettier () {
 }
 
 runKtlint
-runPrettier
+runPrettier || :; # || to avoid cancelling the commit if there is an error with Prettier
 echo "Completed the pre-commit hook."

